### PR TITLE
Adds shuffleGen and shuffleIO, deprecates shuffle', adds a test, updates maintainer.

### PR DESCRIPTION
**Pull request by nicuveo**
That's a lot in one pull request, I can split if you'd prefer!
I had in mind to deprecate shuffle' in 0.0.3, and remove it in 0.1.0.
What do you think?

### CODE REVIEW

File: random-shuffle.cabal [hunk 1]
```diff
@@ -1,12 +1,12 @@
-cabal-version:    >= 1.6
+cabal-version:    >= 1.8
 build-type:       Simple
 name:             random-shuffle
-version:          0.0.2
+version:          0.0.3
```
Comment (exFalso) [inline random-shuffle.cabal]: This bump shouldn't be in the PR (also, it should be 0.0.5, forgot to update the version)

File: src/System/Random/Shuffle.hs [hunk 2]
```diff
@@ -109,23 +112,27 @@ shuffle elements = shuffleTree (buildTree elements)
...
 
 
--- |Given a sequence @(e1,...en)@ to shuffle, its length, and a random
--- generator, compute the corresponding permutation of the input
--- sequence.
-shuffle' :: RandomGen gen => [a] -> Int -> gen -> [a]
-shuffle' elements len = shuffle elements . rseq len
+-- |Given a sequence @(e1,...en)@ to shuffle and a random generator,
+-- compute the corresponding permutation of the input sequence.
+shuffleGen :: RandomGen gen => [a] -> gen -> ([a], gen)
```
Comment (exFalso) [inline src/System/Random/Shuffle.hs:117]: Exactly my thinking. Still unsure why the original `shuffle'` signature contained the length of the list

File: src/System/Random/Shuffle.hs [hunk 3]
```diff
@@ -109,23 +112,27 @@ shuffle elements = shuffleTree (buildTree elements)
...
-            rseq' 0 _ = []
-            rseq' i gen = (j, gen) : rseq' (i - 1) gen'
-                where
-                  (j, gen') = randomR (0, i) gen
+      rseq :: RandomGen gen => Int -> gen -> ([Int], gen)
+      rseq n g = foldl' rnum ([], g) [1..n]
+      rnum (l, g) i = let (j, g') = randomR (0, i) g
+                      in (j : l, g')
+
+{-# DEPRECATED shuffle' "shuffle' is deprecated, please use shuffleGen instead" #-}
```
Comment (exFalso) [inline src/System/Random/Shuffle.hs:130]: :+1:

**Review by exFalso — COMMENTED** [review]
Small thing about version bump, otherwise LGTM

File: random-shuffle.cabal (hunk 1 above)
Comment (nicuveo) [reply random-shuffle.cabal]: I removed the commit.

**Comment by nicuveo** [comment]
Updated. Thanks for the review!

**Review by exFalso — APPROVED** [review]
LGTM 👍

**Comment by alexanderkjeldaas** [comment]
Why can't I file issues in this repo?

Anyways, could there be a sample function for external databases, like if I want to sample rows from an SQL database?

```
shuffleIdx :: RandomGen a => Int -> a -> [Int]
```

**Comment by alexanderkjeldaas** [comment]
@exFalso

**Comment by exFalso** [comment]
@alexanderkjeldaas I think I forked this repo before Github issues were a thing, I enabled them now. Can you please open one? Also how would you like `shuffleIdx` to work?

**Pull request merged**